### ISSUE TITLE
Node v6 can throw startup warnings on OSX under certain configurations

### ISSUE THREAD

**Issue by hetul** [restify/node-restify#1305]
I am on OSX El Capitan 10.11.2, using restify 4.3.0 with Node v6.10.2. When running under this configuration, I get the following errors related to `dtrace-provider` on OSX for my node app:

```
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

Security context: 0x252991dcfb39 <JS Object>#0#
    1: .node [module.js:597] [pc=0x36ccf4ff3e44] (this=0x2f41a4d8b0c1 <an Object with map 0x23f1fd11b2d9>#1#,module=0x16230ded361 <a Module with map 0x23f1fd11c6c9>#2#,filename=0x16230debdd9 <String[101]: /Users/hetul/git/outlier-2/api/node_modules/dtrace-provider/build/Release/DTraceProviderBindings.node>)
    2: load [module.js:~478] [pc=0x36ccf51ae818] (this=0x16230ded361 <a Module with map 0x23f1fd11c6c9>#2#,filename=0x16230debdd9 <String[101]: /Users/hetul/git/outlier-2/api/node_modules/dtrace-provider/build/Release/DTraceProviderBindings.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:446] [pc=0x36ccf4f4145d] (this=0x252991d04381 <undefined>,module=0x16230ded361 <a Module with map 0x23f1fd11c6c9>#2#,filename=0x16230debdd9 <String[101]: /Users/hetul/git/outlier-2/api/node_modules/dtrace-provider/build/Release/DTraceProviderBindings.node>)
    4: _load [module.js:438] [pc=0x36ccf4f35422] (this=0x2f41a4d8b079 <JS Function Module (SharedFunctionInfo 0x2f41a4d399a9)>#3#,request=0x16230dea789 <String[38]: ./build/Release/DTraceProviderBindings>,parent=0x16230de8501 <a Module with map 0x23f1fd11c6c9>#4#,isMain=0x252991d04271 <false>)
    5: require [module.js:~494] [pc=0x36ccf4ff630c] (this=0x16230de8501 <a Module with map 0x23f1fd11c6c9>#4#,path=0x16230dea789 <String[38]: ./build/Release/DTraceProviderBindings>)
    6: require(aka require) [internal/module.js:20] [pc=0x36ccf4f4b8c6] (this=0x252991d04381 <undefined>,path=0x16230dea789 <String[38]: ./build/Release/DTraceProviderBindings>)
... (snipped)
```

This might be caused by restify using an old version `dtrace-provider` which uses an old version of `nan`. I think it this ticket in `dtrace-provider`: https://github.com/chrisa/node-dtrace-provider/issues/74

Generally it works fine. However when another package pulls in an older version of `nan`, `dtrace-provider 0.6.0` just uses the older version of `nan` that is readily available, which causes warnings to be thrown. Upgrading the version of `dtrace-provider` will use better version management which prevents the warning from being thrown.

I have verified that I can fix issues on my local machine by using npm shrinkwrap to force restify to use `dtrace-provider@0.8.1`. (EDIT: of course, the better solution is the upgrade the other dependencies that were causing a conflict)

Here is my `npm-shrinkwrap.json`:
```json
{
  "dependencies": {
    "restify": {
      "version": "4.3.0",
      "from": "restify@>=4.0.0 <5.0.0",
      "dependencies": {
        "dtrace-provider": {
          "version": "^0.8.0",
          "from": "dtrace-provider@>=0.6.0 <0.8.0",
          "optional": true
        }
      }
    }
  }
}
```

While attempting to reproduce this on a smaller test project, I once came a cross a similar error to what is listed in this ticket: https://github.com/restify/node-restify/issues/1284. So that ticket may also be related. Unfortunately, I only saw that once and was unable to reproduce it afterwards.

Here is the package.json for my small project which reproduces this issue:

```json
{
  "name": "repro",
  "version": "1.0.0",
  "description": "Bug repro for restify / dtrace-provider warnings",
  "main": "index.js",
  "dependencies": {
    "restify": "^4.0.0",
    "kafka-node": "^0.3.1"
  }
}
```

And for completeness, a simple index.js:

```js
const restify = require('restify');
const PORT = 8085

let server = restify.createServer();

server.listen(PORT, () => {
    console.log(`Server listening on port ${PORT}`)
})

```

**Comment by retrohacker** [comment]
Hey @hetul,

Thank you so much for taking the time to dive into this issue, the details you provided are deeply appreciated! Do you know if this issue persists in v5 of this project? You can install with `npm install restify/node-restify`.

It sounds like you are pretty close to having a working solution for this, if you can open a PR I would gladly do a review and get it merged :heart:

This sounds like something we will want to backport to v4 too until v5 gets out of beta.

**Comment by retrohacker** [comment]
P.S. I'm marking this as unconfirmed for the moment until a maintainer can reproduce it locally

**Comment by hetul** [comment]
Thanks @retrohacker 

I can confirm I still see the issue on 5.x.

I'll put up a PR to address the issue. It should be easy - the version dtrace-provider just needs to be bumped up.

I'll do my best to run the tests locally, but a maintainer will probably have to make sure it actually works!

**Comment by hetul** [comment]
@retrohacker - I have submitted the PR.

BTW - this is completely irrelevant, but when attempting to install 5.x, your original instructions were incorrect (I notice you edited your post to correct it!). It was because NPM did not respect the default branch set in github. I opened an issue in NPM here:  https://github.com/npm/npm/issues/16453

Let me know if you already knew of that issue and there exists something tracking it in NPM, and I can close my issue there. Thanks!
